### PR TITLE
Add support for loading namespaces from pod configuration.

### PR DESCRIPTION
**Pull request by brendandburns**
See #636

### CODE REVIEW

File: src/KubernetesClient/FileUtils.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+using System;
+using System.IO.Abstractions;
+
+namespace k8s
+{
+    public static class FileUtils
```
Comment (tg123) [inline src/KubernetesClient/FileUtils.cs:6]:
```suggestion
    internal static class FileUtils
```

Comment (tg123) [reply src/KubernetesClient/FileUtils.cs:6]: or I think make `IFileSystem` a field of config object which would be better than global static.
but for test purpose, it is good.

**Review by tg123 — APPROVED** [review]

**Comment by tg123** [comment]
some nits then /LGTM

**Comment by brendandburns** [comment]
Committed your suggestion. Thanks!

**Comment by brendandburns** [comment]
@tg123 any ideas on the test failure here? This e2e test passes on my windows machine.

**Comment by brendandburns** [comment]
Hrm, now if failed on ubuntu. I will try to reproduce the issue.

**Comment by tg123** [comment]
flaky  test
re-run passed

**Comment by brendandburns** [comment]
Thanks we should de-flake that test.

**Comment by tg123** [comment]
> Thanks we should de-flake that test.

sure i will take care of the flake tests.
The test is now counting how many heartbeats in a fixed time period, would fail if testing machine is slow and did not emit enough heartbeat.

**Comment by brendandburns** [comment]
I'm happy to take a look at the test if you prefer, let me know.

**Comment by tg123** [comment]
> I'm happy to take a look at the test if you prefer, let me know.

sure appreciate it
tests in 
LeaderElectionTests.cs

**Comment by brendandburns** [comment]
Ok, I will look to deflake those tests.

**Pull request merged**